src/radical/pilot/utils/component.py

```diff
@@ -584,8 +594,13 @@ def _subscriber(q, callback):
         q = rpu_Pubsub.create(rpu_PUBSUB_ZMQ, pubsub, rpu_PUBSUB_SUB, addr)
         q.subscribe(topic)
 
-        t = mt.Thread(target=_subscriber, args=[q,cb],
-                      name="%s.subscriber" % self.cname)
+        if topic in os.getenv("RADICAL_PILOT_CPROFILE_SUBSCRIBERS", "").split():
+            t = ProfiledThread(target=_subscriber, args=[q,cb],
+                               name="%s-%s.subscriber" % (self.cname, topic))
+        else:
+            t = mt.Thread(target=_subscriber, args=[q,cb],
+                          name="%s.subscriber" % self.cname)
+
```

Comment:
IIRC, I was missing out on most of the activity by only profiling the components directly. With this code, all of the activity in the subscriber threads is captured too. Makes sense?